compiler/rustc_symbol_mangling/src/v0.rs

```diff
@@ -32,6 +32,7 @@ pub(super) fn mangle<'tcx>(
     let mut cx: SymbolMangler<'_> = SymbolMangler {
         tcx,
         start_offset: prefix.len(),
+        is_exportable: tcx.is_exportable(def_id),
```

Comment:
Maybe pass this as an argument to `fn mangle`, we already know whether it's exportable when calling this.